src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -134,7 +134,11 @@ public void AttachProfiler(TimeSpan attachTimeout, Guid profilerGuid, string pro
             switch ((DiagnosticsServerResponseId)response.Header.CommandId)
             {
                 case DiagnosticsServerResponseId.Error:
-                    var hr = BitConverter.ToInt32(response.Payload, 0);
+                    uint hr = BitConverter.ToUInt32(response.Payload, 0);
+                    if (hr == (uint)DiagnosticsIpcError.UnknownCommand) 
+                    {
+                      throw new InvalidOperationException("The target runtime does not support profiler attach.");
```

Comment:
I think we should avoid throwing new exception types that don't derive from DiagnosticsClientException. Ideally this code would always be safe for the caller assuming they call with valid arguments:
````C#
try  { client.DoSomeCommand(...) } 
catch (DiagnosticsClientException e)
{
   Log(e);
}
````

 Depending on whether you want to be able to catch this case in isolation I'd suggest either:
````C#
    throw new ServerErrorException("The target runtime does not support profiler attach");
````
or
````C#
    // this requires creating a new class UnsupportedCommandException : ServerErrorException { ... }
    throw new UnsupportedCommandException("The target runtime does not support profiler attach");
````

cc @sywhang @josalem

Reply:
Hi @noahfalk , `UnsupportedCommandException` is better on my opinion. I'll make the fix. However, the current version of code throws `PlatformNotSupportedException` in the dump command. Should I fix this case too?

Reply:
> Should I fix this case too?

Yeah that the spot I noticed already had this issue. I'd be glad to accept it if you are up for a little more work, either as part of this PR or in a separate PR. I think you would also need to change some code in dotnet-dump that checks for the PNSE exception type and some unit tests that assert that PNSE is being thrown. cc @mikem8361.